kyo-prelude/shared/src/main/scala/kyo2/Abort.scala

```diff
@@ -83,23 +83,25 @@ object Abort:
                     )
                 }
             } {
-                case fail: E => Result.fail(fail)
-                case fail    => Result.panic(fail)
+                case fail: E if classOf[Throwable].isAssignableFrom(ct.runtimeClass) =>
```

Comment:
@hearnadam I think you had warned about this 🙈 I'm using `Abort` for the new `Fiber` implementation and it's incorrectly translating thrown exceptions that should be a `Panic` to `Fail` instead. I could work it around in `Fiber` but this seems a safer behavior.

Reply:
Can you try using `if tag <:< Tag[Throwable]` to check if the result should be a `fail`?

Reply:
Yep, tests pass with it as well! It's hard to tell if it'd have better performance than `isAssignableFrom`, though. I'd prefer to keep it as is since the `Tag` in this method is more of a workaround.